### DISCUSSION TITLE
Workaround for 'MainPage' does not contain a definition for 'InitializeComponent' and The name 'x' does not exist in the current context intellisense cache errors

### DISCUSSION THREAD

**Discussion by Rytisgit** [unoplatform/uno#11368] [General]
Having intellisense problems all the time while developing my wasm app prompted me to look into a way to fix them and I found something promising: https://github.com/dotnet/roslyn/issues/44093

I tested out the workaround presented in the issue description:
```
  <PropertyGroup>
    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
  </PropertyGroup>

  <ItemGroup>
    <Compile Include="$(CompilerGeneratedFilesOutputPath)\**" />
  </ItemGroup>
 
  <Target Name="RemoveSourceGeneratedFiles" BeforeTargets="CoreCompile">
    <ItemGroup>
      <Compile Remove="$(CompilerGeneratedFilesOutputPath)\**" />
    </ItemGroup>
  </Target>
```

Adding these lines to my wasm .csproj solved the issues and now no more errors are shown and autocomplete is available. As https://github.com/dotnet/roslyn/issues/63224 is still not closed, some version of this could be included in the nugets to provide working intellisense.

**Comment by jeromelaban** [comment]
Thanks for looking into this! Have you tried running with the latest templates, which use the a separate project and not a shared project?

**Reply by Rytisgit** [reply]
I hadn't tested when I posted it, I tested the new templates now. This method does manage to fix the the intellisense errors in the new templates as well, when including the code in the separate shared project on rebuild. But another problem exists when using it - while previously intellisense never worked on edit, save, or restart of VS, now in the new templates the file is generated just once on starting the visual studio project, and then with this fix another file is updated on every rebuild as well, causing the ambiguous reference errors for any parameters that were present on start due to 2 separate files existing:
![image](https://user-images.githubusercontent.com/36188103/220150129-1da9a32f-dc73-444e-8200-982522fefc44.png)
 
I think if the never updated file could be not generated, or deleted it could be a usable solution.

Also I noticed the build slowing down quite a bit with this included in the new projects, I had gotten used to my wasm being very slow so I didn't notice before. It could be a downside of this, or maybe there's some better similar solution in the other roslyn discussion.

I also had a single instance of the build refusing to complete due to ambiguous errors or same declarations while testing, when the project has both net7.0-windows10.0.18362 and net7.0 but I can't replicate it. it never happened when the project had only the net7.0 option.

**Reply by jeromelaban** [reply]
Thank for the update. Indeed the build is slower because of the cross targeting. This is a known [Visual Studio issue](https://developercommunity.visualstudio.com/t/Building-a-cross-targeted-project-with-m/651372?space=8&q=building-a-cross-targeted-project-with-many-target) (please upvote it!) and you can [use this technique](https://platform.uno/docs/articles/guides/solution-building-single-targetframework.html) to make the build go faster.
